// react-bootstrap/cjs/Accordion.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _classnames = _interopRequireDefault(require("classnames"));

var React = _interopRequireWildcard(require("react"));

var _uncontrollable = require("uncontrollable");

var _ThemeProvider = require("./ThemeProvider");

var _AccordionBody = _interopRequireDefault(require("./AccordionBody"));

var _AccordionButton = _interopRequireDefault(require("./AccordionButton"));

var _AccordionCollapse = _interopRequireDefault(require("./AccordionCollapse"));

var _AccordionContext = _interopRequireDefault(require("./AccordionContext"));

var _AccordionHeader = _interopRequireDefault(require("./AccordionHeader"));

var _AccordionItem = _interopRequireDefault(require("./AccordionItem"));

var _jsxRuntime = require("react/jsx-runtime");

var Accordion = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _useUncontrolled = (0, _uncontrollable.useUncontrolled)(props, {
    activeKey: 'onSelect'
  }),
      _useUncontrolled$as = _useUncontrolled.as,
      Component = _useUncontrolled$as === void 0 ? 'div' : _useUncontrolled$as,
      activeKey = _useUncontrolled.activeKey,
      bsPrefix = _useUncontrolled.bsPrefix,
      className = _useUncontrolled.className,
      onSelect = _useUncontrolled.onSelect,
      flush = _useUncontrolled.flush,
      controlledProps = (0, _objectWithoutPropertiesLoose2.default)(_useUncontrolled, ["as", "activeKey", "bsPrefix", "className", "onSelect", "flush"]);

  var prefix = (0, _ThemeProvider.useBootstrapPrefix)(bsPrefix, 'accordion');
  var contextValue = (0, React.useMemo)(function () {
    return {
      activeEventKey: activeKey,
      onSelect: onSelect
    };
  }, [activeKey, onSelect]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_AccordionContext.default.Provider, {
    value: contextValue,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(Component, (0, _extends2.default)({
      ref: ref
    }, controlledProps, {
      className: (0, _classnames.default)(className, prefix, flush && prefix + "-flush")
    }))
  });
});
Accordion.displayName = 'Accordion';

var _default = Object.assign(Accordion, {
  Button: _AccordionButton.default,
  Collapse: _AccordionCollapse.default,
  Item: _AccordionItem.default,
  Header: _AccordionHeader.default,
  Body: _AccordionBody.default
});

exports.default = _default;
module.exports = exports.default;
